Tests/Algorithm/AlgorithmRegisterIndicatorTests.cs

```diff
@@ -0,0 +1,157 @@
+
+            _indicatorTestsGroups =
+                from type in GetType().Assembly.GetTypes()
+                where type.IsPublic && !type.IsAbstract
+                where
+                   typeof(CommonIndicatorTests<TradeBar>).IsAssignableFrom(type) ||
+                   typeof(CommonIndicatorTests<IBaseDataBar>).IsAssignableFrom(type) ||
+                   typeof(CommonIndicatorTests<IndicatorDataPoint>).IsAssignableFrom(type)
+                group type by type.BaseType.GetGenericArguments().FirstOrDefault();
+        }
+
+        [Test]
+        public void RegistersIndicatorProperly()
+        {
+            var expected = 0;
+            foreach (var group in _indicatorTestsGroups)
+            {
+                var key = group.Key;
+                foreach (var type in group)
+                {
+                    var indicatorTest = Activator.CreateInstance(type);
+                    if (key == typeof(IndicatorDataPoint))
+                    {
+                        var indicator = (indicatorTest as CommonIndicatorTests<IndicatorDataPoint>).Indicator;
+                        Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute, Field.Close));
+                        expected++;
+                    }
+                    else if (key == typeof(IBaseDataBar))
+                    {
+                        var indicator = (indicatorTest as CommonIndicatorTests<IBaseDataBar>).Indicator;
+                        Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
+                        expected++;
+                    }
+                    else if (key == typeof(TradeBar))
+                    {
+                        var indicator = (indicatorTest as CommonIndicatorTests<TradeBar>).Indicator;
+                        Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
+                        expected++;
+                    }
+                    var actual = _algorithm.SubscriptionManager.Subscriptions.FirstOrDefault().Consolidators.Count;
+                    Assert.AreEqual(expected, actual);
+                }
+            }
+        }
+
+        [Test, Ignore]
+        public void RegistersIndicatorProperlyPython()
+        {
+            var expected = 0;
+            foreach (var group in _indicatorTestsGroups)
+            {
+                var key = group.Key;
+                foreach (var type in group)
+                {
+                    var indicatorTest = Activator.CreateInstance(type);
+                    if (key == typeof(IndicatorDataPoint))
+                    {
+                        using (Py.GIL())
+                        {
+                            var indicator = (indicatorTest as CommonIndicatorTests<IndicatorDataPoint>).Indicator.ToPython();
+                            Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
```

Comment:
I love what's going on here but would like us to clean it up a little bit so we can use it in the future. Can we add a method to `CommonIndicatorTests<T>` to get the `PyObject`, maybe something like:
```csharp
public PyObject GetIndicatorAsPyObject()
```
which would handle getting the indicator instance and invoking `ToPython()` to convert it to `PyObject`.

With the above in place, I think this entire if/else block collapses and will make the test much clearer.

thoughts?

Reply:
Agree and made some changes. 
I couldn't remove all if/else block, because we of generics, so it is following `CommonIndicatorTests` pattern.